src/integrationTest/samVisualize/visualizeSamTemplate.test.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as fs from 'fs-extra'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
+import { spy } from 'sinon'
+import { closeAllEditors } from '../../shared/utilities/vsCodeUtils'
+
+const sampleSamTemplateFileName = 'helloWorld.yaml'
+const sampleSamTemplateYaml = `
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: >
+    samp-app
+    Sample SAM Template for samp-app
+
+Resources:
+    HelloWorldFunction:
+        Type: AWS::Serverless::Function 
+        Properties:
+            CodeUri: hello-world/
+            Handler: app.lambdaHandler
+            Runtime: nodejs14.x
+            Events:
+                HelloWorld:
+                    Type: Api
+                    Properties:
+                        Path: /hello
+                        Method: get`
+/**
+ * Helper function to create a text file with supplied content and open it with a TextEditor
+ *
+ * @param fileText The supplied text to fill this file with
+ * @param fileName The name of the file to save it as. Include the file extension here.
+ *
+ * @returns TextEditor that was just opened
+ */
+async function openATextEditorWithText(fileText: string, fileName: string): Promise<vscode.TextEditor> {
```

Comment:
this already exists in `src/integrationTest/stepFunctions/visualizeStateMachine.test.ts` . also seems like it could be generalized, it is not really test-specific.

Reply:
I'll move it to `src/shared/utilities/textDocumentUtilities.ts`. I'll also have to add the `folder` as a parameter, since it depends on a global variable (`tempFolder`)